shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -1293,24 +1294,30 @@ routing message to a random node so that:
                     body: ResponseBody::Fetch(fetch),
                     ..
                 })) => {
-                    for response in &fetch.responses {
-                        for partition in &response.partitions {
+                    // Clear this optional field to avoid making clients try to bypass shotover
+                    // partition.current_leader and partition.preferred_read_replica are cleared due to the same reason
+                    fetch.node_endpoints.clear();
+                    for fetch_response in &mut fetch.responses {
+                        for partition in &mut fetch_response.partitions {
+                            partition.current_leader = LeaderIdAndEpoch::default();
```

Comment:
While setting current_leader.leader_id to -1 is good.
I'm not sure if current_leader.leader_epoch should be set to -1 or left as is.
I think that drivers encountering a -1 value here would just ignore it, but if the DB always populates this on routing error then the driver might assume its populated.

In conclusion, I think that what you have is fine.

Reply:
I'm happy to change this line to rewrite current_leader.leader_id to -1 while keeping current_leader.leader_epoch unchanged, if that's a safer approach.

Reply:
I think what you have is the better approach, just noting the possibility here in case we do hit issues down the road.